tests/rados/test_scrub_parameters.py

```diff
@@ -0,0 +1,252 @@
+    if not res:
+        log.error("PG's in cluster are not active + Clean ")
+        return 1
+    osd_list = rados_object.get_active_osd_list()
+    osd_id = random.choice(osd_list)
+    return osd_id
+
+
+def chk_forced_flag(scrub_object, osd_id):
+    """
+    Method is used to check the forced flag status
+    Args:
+        rados_object: Rados object
+        osd_id: OSD ID number
+    Returns:
+        True -> If forced flag is true
+        False -> If forced flag is false
+    """
+    pg_id = ""
+    time_execution = datetime.datetime.now() + datetime.timedelta(minutes=20)
+    found = False
+    while datetime.datetime.now() < time_execution:
+        out_put = scrub_object.get_dump_scrubs(osd_id)
+        for pg_no in out_put:
+            if pg_no["forced"] is True:
+                pg_id = pg_no["pgid"]
+                log.info(f"The forced flag is true for the pg id-{pg_id}")
+                found = True
+                return True, pg_id
+        log.info("Scrub/Deep-scrub verification is in-progress")
+        time.sleep(30)
+    if not found:
+        return False, pg_id
+
+
+def wait_pg_forced_flag(scrub_object, osd_id, pg_id):
+    """
+    Method is used to wait for the forced flag status
+    Args:
+        scrub_object: Scrub object
+        osd_id: OSD ID number
+        pg_id: pg id number
+    Returns:
+        True -> If forced flag set to  false
+        False -> If forced flag is true
+    """
+    time_execution = datetime.datetime.now() + datetime.timedelta(minutes=60)
+    while datetime.datetime.now() < time_execution:
+        pg_dump_scrub = scrub_object.get_pg_dump_scrub(osd_id, pg_id)
+        log.info(f"The {pg_id} dumb scrub is - {pg_dump_scrub}")
+        if pg_dump_scrub["forced"] is False:
+            log.info("The forced flag is set to False")
+            return True
+        log.info(
+            f"The forced flag for the {pg_id} is True and scrubbing is in progress"
+        )
+        time.sleep(30)
+    log.error("The forced flag is not set to True")
+    return False
+
+
+def verify_scheduled_time(mon_object, prev_dump_scrub, pg_id, pg_current_sched_time):
+    """
+    Method is used to verify the scheduled time is less than the  osd_scrub_min_interval+
+    (osd_scrub_min_interval * osd_scrub_interval_randomized_ratio)
+    Args:
+        mon_object: Monitor object
+        prev_dump_scrub: dumb scrub before scrub/deep-scrub operation
+        pg_id: pg id number
+        pg_current_sched_time: current scheduled time
+    Returns:
+        True ->  if scheduled time <=  osd_scrub_min_interval+
+    (osd_scrub_min_interval * osd_scrub_interval_randomized_ratio)
+        False -> if scheduled time >  osd_scrub_min_interval+
+    (osd_scrub_min_interval * osd_scrub_interval_randomized_ratio)
+
+    """
+    pg_prev_sched_time = ""
+    for pg_no in prev_dump_scrub:
+        if pg_no["pgid"] == pg_id:
+            pg_prev_sched_time = pg_no["sched_time"]
```

Comment:
need to return fail in the method if `pg_prev_sched_time` could not be set.

Reply:
Added the code.